### ISSUE TITLE
ERROR malloc(): invalid next->prev_inuse (unsorted)

### ISSUE THREAD

**Issue by lianglilong-gloritytech** [OML-Team/open-metric-learning#426]
I'm sorry that I have encountered some problems in use, could you please help me?

When I train the CUB dataset in ubuntu using the following command:
+ CUDA_VISIBLE_DEVICES=2
+ python3 train_cub.py bs_val=8 max_epochs=10 optimizer.args.lr=0.0001 dataset_root=/root/workspace/dataset_dir/CUB_200_2011 num_workers=2

I get the following error:
![image](https://github.com/OML-Team/open-metric-learning/assets/134380502/3bb94f37-f3e3-46f9-a264-da784f84f286)

The same problem will occur when I validate：
![image](https://github.com/OML-Team/open-metric-learning/assets/134380502/7b57a27e-e632-48a2-b2c3-dbe244198f98)

In fact, this error occurred once during training, but I changed the num_workers to a smaller size and it did not appear.
![image](https://github.com/OML-Team/open-metric-learning/assets/134380502/9d478a08-7e0f-4b03-8c94-cd6889a9f17f)

I now suspect that my memory is still too small? But I think I set it small enough. . So I would like you to help me to see what is the problem? Thank you so much!!!

In addition, I would like to ask, I want to use the CUB data set to train my own model, is there any wrong with this training? Is there any step missing? Please correct me, thank you thank you

**Comment by AlekseySh** [comment]
Hi, @lianglilong-gloritytech and thank you for using OML!

I guess that the error is not OML-specific. Did you train other models in the DDP regime before using PyTorch Lightning or other frameworks?

> In fact, this error occurred once during training, but I changed the num_workers to a smaller size and it did not appear.

How many workers did you have in your smaller setup -- two? I'm asking because there is a big difference between 0 and 1,2,3...

PS. Are you sure you need DDP for CUB dataset? It's tiny

**Comment by AlekseySh** [comment]
> In addition, I would like to ask, I want to use the CUB data set to train my own model, is there any wrong with this training? Is there any step missing? Please correct me, thank you thank you

Except for the technical memory problem, I think everything is fine with your pipeline because you obtained cmc@1 = 83.87. If you check the Zoo table in the readme you may see that it's close to what we obtained (86.6). So, from the OML point of view, you are good to train your custom model.

**Comment by AlekseySh** [comment]
@DaloroAT did you see a problem like that before?

**Comment by lianglilong-gloritytech** [comment]
> > In fact, this error occurred once during training, but I changed the num_workers to a smaller size a

**Comment by lianglilong-gloritytech** [comment]
Sorry for accidentally closing the question. . Now I reopen it. .

> I guess that the error is not OML-specific. Did you train other models in the DDP regime before using PyTorch Lightning or other frameworks?

Forehead. Yes, I ran the deepglint/unicom model with DDP before in the same container. But I gave up when I encountered some problems that I couldn't solve.

> How many workers did you have in your smaller setup -- two? I'm asking because there is a big difference between 0 and 1,2,3...

In fact, changing num_workers to 2 is just a tentative change after reading here. Because the original yaml file is 20, I am worried that the memory is not enough
https://github.com/OML-Team/open-metric-learning/issues/364
As a novice, I would like to ask for advice. What is the difference between 0 and 1,2,3.., and how to set num_workers more reasonably?

> PS. Are you sure you need DDP for CUB dataset? It's tiny

Of course not, in fact, I just need to use the CUB dataset to train my own metric learning model. In fact, there is really no need to use DDP

Thanks

**Comment by DaloroAT** [comment]
Hey @lianglilong-gloritytech 
Could you try to place `torch.backends.mkldnn.enabled = False` at the beginning of your training script?

**Comment by DaloroAT** [comment]
Does it appear only when using DDP? Or happens always?

**Comment by AlekseySh** [comment]
In general, you can set num workers as the number of your CPU cores (with hyperthreading) minus 1 or two. 
But it's better to try several values and compare the time needed to process one batch.

**Comment by AlekseySh** [comment]
You can also try `--shm-size` when running your image.

But yes, from what you said, it sound's like a non-OML issue.

**Comment by lianglilong-gloritytech** [comment]
@AlekseySh @DaloroAT 
Thanks Reply. I think I found the possible reason: logger uses Neptune.
I changed the logger to the following code:
`logger:`
`  name: tensorboard`
`    args:`
`      save_dir: "."`

no more errors occur
I think it might be a code error inside Neptune

Thanks for the help. I will try to train the model first, if there is any problem, I will ask for your help.

**Comment by DaloroAT** [comment]
Unexpected source for such error 😁

**Comment by AlekseySh** [comment]
@lianglilong-gloritytech how did you understand that the problem may be in Neptune?

**Comment by lianglilong-gloritytech** [comment]
> @lianglilong-gloritytech how did you understand that the problem may be in Neptune?

It can only be said that the error may have occurred inside Neptune.
The reason I think so is because I don't have this problem after replacing the logger.
It may be the wrong way to use it, because I didn't set NEPTUNE_API_TOKEN, and I used the default train_cub.yaml directly

**Comment by AlekseySh** [comment]
@lianglilong-gloritytech hmmm, interesting

Thanks to you I've create a PR to produce better error message when token is not specified: https://github.com/OML-Team/open-metric-learning/pull/431
